prow/plugins/lifecycle/lifecycle_test.go

```diff
@@ -17,133 +17,248 @@ limitations under the License.
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{lifecycleRottenLabel},
 		},
 		{
-			body:  "/lifecycle stale",
-			added: []string{"lifecycle/stale"},
+			name:                  "remove lifecycle frozen, have it",
+			body:                  "/remove-lifecycle frozen",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleFrozenLabel},
+			issueLabels:           []string{lifecycleFrozenLabel},
 		},
 		{
-			body:  "/lifecycle putrid",
-			added: []string{"lifecycle/putrid"},
+			name:                  "remove lifecycle frozen, don't have it",
+			body:                  "/remove-lifecycle frozen",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body:  "/lifecycle rotten",
-			added: []string{"lifecycle/rotten"},
+			name:                  "remove lifecycle stale, have it",
+			body:                  "/remove-lifecycle stale",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleStaleLabel},
+			issueLabels:           []string{lifecycleStaleLabel},
 		},
 		{
-			body:    "/remove-lifecycle frozen",
-			removed: []string{"lifecycle/frozen"},
+			name:                  "remove lifecycle stale, don't have it",
+			body:                  "/remove-lifecycle stale",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body:    "/remove-lifecycle stale",
-			removed: []string{"lifecycle/stale"},
+			name:                  "remove lifecycle rotten, have it",
+			body:                  "/remove-lifecycle rotten",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleRottenLabel},
+			issueLabels:           []string{lifecycleRottenLabel},
 		},
 		{
-			body:    "/remove-lifecycle stale\n/remove-lifecycle putrid\n/lifecycle frozen",
-			added:   []string{"lifecycle/frozen"},
-			removed: []string{"lifecycle/stale", "lifecycle/putrid"},
+			name:                  "remove lifecycle rotten, don't have it",
+			body:                  "/remove-lifecycle rotten",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
+		},
+		{
+			name:                  "add lifecycle frozen and remove stale and rotten, have stale and rotten",
+			body:                  "/lifecycle frozen",
+			expectedNewLabels:     []string{lifecycleFrozenLabel},
+			expectedRemovedLabels: []string{lifecycleStaleLabel, lifecycleRottenLabel},
+			issueLabels:           []string{lifecycleStaleLabel, lifecycleRottenLabel},
+		},
+		{
+			name:                  "remove lifecycle frozen, have all",
+			body:                  "/remove-lifecycle frozen",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{lifecycleFrozenLabel},
+			issueLabels:           []string{lifecycleFrozenLabel, lifecycleStaleLabel, lifecycleRottenLabel},
```

Comment:
Is this a valid state? Should an issue/PR have Stale and Rotten too?

Reply:
> Is this a valid state? Should an issue/PR have Stale and Rotten too?

This is not a valid state. If automation is used to add labels after this, only one lifecycle label will exist at a time.

However, I think we have PRs in the wild with more than one lifecycle label right now. This test takes care of that.

It doesn't matter once the automation is added though -- because such a case cannot exist then.

Reply:
Certainly a possible state. PRs being both frozen and stale are pretty common. Also there's no guarantee that both the remove stale and add rotten rpcs both pass. We could wind up with all possible states depending on how github feels today 😁

Reply:
Sorry didn't see you already replied to this one